### PR TITLE
more MSA things

### PR DESCRIPTION
**Pull request by jamesmkrieger**

### CODE REVIEW

File: prody/ensemble/functions.py [hunk 1]
```diff
@@ -364,18 +364,40 @@ def buildPDBEnsemble(refpdb, PDBs, title='Unknown', labels=None, seqid=94, cover
...
+    if alignments_list is None:
+        alignments_list = [None]* len(PDBs)
+    else:
+        if len(alignments_list) != len(PDBs):
+            raise TypeError('alignments_list and PDBs must have the same lengths.')
 
     # obtain the hierarhical view of the reference PDB
     refhv = refpdb.getHierView()
-    refchains = list(refhv)
+    refchains = refhv.iterChains()
```
Comment (SHZ66) [inline prody/ensemble/functions.py]: Using iterChains will cause an error on the next line.

File: prody/proteins/compare.py [hunk 2]
```diff
@@ -17,8 +17,7 @@
     range = xrange
 
 __all__ = ['matchChains', 'matchAlign', 'mapOntoChain',
-           'mapChainByChain', 'mapOntoChainByAlignment', 
-           'getMatchScore', 'setMatchScore',
+           'mapChainByChain', 'getMatchScore', 'setMatchScore',
```
Comment (SHZ66) [inline prody/proteins/compare.py]: mapOntoChainByAlignment was added in #537

File: prody/proteins/compare.py [hunk 3]
```diff
@@ -966,26 +965,6 @@ def mapChainByChain(atoms, ref, **kwargs):
...
-        alignment = alignments[index]
-        tar_aligned_seq = alignment[-1]
-        hv = atoms.getHierView()
-        for target_chain in hv.iterChains():
-            tar_seq = target_chain.getSequence()
-            if tar_seq == tar_aligned_seq.replace('-', ''):
-                mappings = mapOntoChain(target_chain, chain, alignment=alignment, **kwargs)
-                return mappings
-        LOGGER.warn('The sequence of chain does not match that in alignment (index = %d).'%index)
-    return []
```
Comment (SHZ66) [inline prody/proteins/compare.py]: mapOntoChainByAlignment was added in #537

File: prody/ensemble/functions.py [hunk 4]
```diff
@@ -364,18 +364,40 @@ def buildPDBEnsemble(refpdb, PDBs, title='Unknown', labels=None, seqid=94, cover
...
+    else:
+        if len(alignment_list) != len(PDBs):
+            raise TypeError('alignment_list and PDBs must have the same lengths.')
+
+    alignments_list = kwargs.get('alignments_list', None)
+    if alignments_list is None:
+        alignments_list = [None]* len(PDBs)
+    else:
+        if len(alignments_list) != len(PDBs):
+            raise TypeError('alignments_list and PDBs must have the same lengths.')
```
Comment (SHZ66) [inline prody/ensemble/functions.py]: alignment_list is wrapped in kwargs so it does not need to be processed. See #537

File: prody/ensemble/functions.py (hunk 1 above)
Comment (jamesmkrieger) [reply prody/ensemble/functions.py]: oh ok

File: prody/proteins/compare.py (hunk 3 above)
Comment (jamesmkrieger) [reply prody/proteins/compare.py]: Again I don't know what happened here.

File: prody/ensemble/functions.py (hunk 4 above)
Comment (jamesmkrieger) [reply prody/ensemble/functions.py]: ok

File: prody/proteins/compare.py (hunk 2 above)
Comment (jamesmkrieger) [reply prody/proteins/compare.py]: This got deleted by accident. I'm restoring it back

File: prody/ensemble/functions.py (hunk 4 above)
Comment (jamesmkrieger) [reply prody/ensemble/functions.py]: It still needs documenting though I guess

Comment (SHZ66) [reply prody/ensemble/functions.py]: yes sure

**Pull request merged**